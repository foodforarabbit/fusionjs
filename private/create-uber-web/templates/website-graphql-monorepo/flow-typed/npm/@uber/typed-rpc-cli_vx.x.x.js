// flow-typed signature: 4996af5695c1d82b2b06dd8b263ba2de
// flow-typed version: <<STUB>>/@uber/typed-rpc-cli_v^1.7.3/flow_v0.90.0

/**
 * This is an autogenerated libdef stub for:
 *
 *   '@uber/typed-rpc-cli'
 *
 * Fill this stub out by replacing all the `any` types.
 *
 * Once filled out, we encourage you to share your work with the
 * community by sending a pull request to:
 * https://github.com/flowtype/flow-typed
 */

declare module '@uber/typed-rpc-cli' {
  declare module.exports: any;
}

/**
 * We include stubs for each file inside this npm package in case you need to
 * require those files directly. Feel free to delete any files that aren't
 * needed.
 */
declare module '@uber/typed-rpc-cli/bin/cli' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/__tests__/config.test' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/__tests__/convert-case.test' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/__tests__/multiple-web-rpc-target.test' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/__tests__/multiple.test' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/__tests__/populous.test' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/__tests__/replace-extension.test' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/__tests__/staging.test' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/add-comment' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/babel-opts' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/babel-plugin-transform-tree-shake' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/convert-case' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/ensure-import-declaration' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/file-utils' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/generate-graphql-code' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/generate-plugin-code' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/generate-plugin-register-code' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/generate-register-graphql-links' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/generate-register-graphql-plugins' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/generate-resolvers' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/generate-rpc-code' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/generate-schemas' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/generate-type-imports' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/generate' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/get-named-exports' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/get-services' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/get-thrift-files' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/get-unique-token-plugin-pair' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/index' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/link' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/load-config' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/main' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/mod-flow' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/parse-config' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/remove-unused-imports' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/replace-extension' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/test-utils/run-test' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/update-flow' {
  declare module.exports: any;
}

declare module '@uber/typed-rpc-cli/src/update-main-schema' {
  declare module.exports: any;
}

// Filename aliases
declare module '@uber/typed-rpc-cli/bin/cli.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/bin/cli'>;
}
declare module '@uber/typed-rpc-cli/src/__tests__/config.test.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/__tests__/config.test'>;
}
declare module '@uber/typed-rpc-cli/src/__tests__/convert-case.test.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/__tests__/convert-case.test'>;
}
declare module '@uber/typed-rpc-cli/src/__tests__/multiple-web-rpc-target.test.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/__tests__/multiple-web-rpc-target.test'>;
}
declare module '@uber/typed-rpc-cli/src/__tests__/multiple.test.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/__tests__/multiple.test'>;
}
declare module '@uber/typed-rpc-cli/src/__tests__/populous.test.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/__tests__/populous.test'>;
}
declare module '@uber/typed-rpc-cli/src/__tests__/replace-extension.test.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/__tests__/replace-extension.test'>;
}
declare module '@uber/typed-rpc-cli/src/__tests__/staging.test.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/__tests__/staging.test'>;
}
declare module '@uber/typed-rpc-cli/src/add-comment.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/add-comment'>;
}
declare module '@uber/typed-rpc-cli/src/babel-opts.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/babel-opts'>;
}
declare module '@uber/typed-rpc-cli/src/babel-plugin-transform-tree-shake.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/babel-plugin-transform-tree-shake'>;
}
declare module '@uber/typed-rpc-cli/src/convert-case.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/convert-case'>;
}
declare module '@uber/typed-rpc-cli/src/ensure-import-declaration.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/ensure-import-declaration'>;
}
declare module '@uber/typed-rpc-cli/src/file-utils.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/file-utils'>;
}
declare module '@uber/typed-rpc-cli/src/generate-graphql-code.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/generate-graphql-code'>;
}
declare module '@uber/typed-rpc-cli/src/generate-plugin-code.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/generate-plugin-code'>;
}
declare module '@uber/typed-rpc-cli/src/generate-plugin-register-code.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/generate-plugin-register-code'>;
}
declare module '@uber/typed-rpc-cli/src/generate-register-graphql-links.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/generate-register-graphql-links'>;
}
declare module '@uber/typed-rpc-cli/src/generate-register-graphql-plugins.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/generate-register-graphql-plugins'>;
}
declare module '@uber/typed-rpc-cli/src/generate-resolvers.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/generate-resolvers'>;
}
declare module '@uber/typed-rpc-cli/src/generate-rpc-code.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/generate-rpc-code'>;
}
declare module '@uber/typed-rpc-cli/src/generate-schemas.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/generate-schemas'>;
}
declare module '@uber/typed-rpc-cli/src/generate-type-imports.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/generate-type-imports'>;
}
declare module '@uber/typed-rpc-cli/src/generate.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/generate'>;
}
declare module '@uber/typed-rpc-cli/src/get-named-exports.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/get-named-exports'>;
}
declare module '@uber/typed-rpc-cli/src/get-services.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/get-services'>;
}
declare module '@uber/typed-rpc-cli/src/get-thrift-files.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/get-thrift-files'>;
}
declare module '@uber/typed-rpc-cli/src/get-unique-token-plugin-pair.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/get-unique-token-plugin-pair'>;
}
declare module '@uber/typed-rpc-cli/src/index.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/index'>;
}
declare module '@uber/typed-rpc-cli/src/link.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/link'>;
}
declare module '@uber/typed-rpc-cli/src/load-config.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/load-config'>;
}
declare module '@uber/typed-rpc-cli/src/main.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/main'>;
}
declare module '@uber/typed-rpc-cli/src/mod-flow.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/mod-flow'>;
}
declare module '@uber/typed-rpc-cli/src/parse-config.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/parse-config'>;
}
declare module '@uber/typed-rpc-cli/src/remove-unused-imports.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/remove-unused-imports'>;
}
declare module '@uber/typed-rpc-cli/src/replace-extension.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/replace-extension'>;
}
declare module '@uber/typed-rpc-cli/src/test-utils/run-test.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/test-utils/run-test'>;
}
declare module '@uber/typed-rpc-cli/src/update-flow.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/update-flow'>;
}
declare module '@uber/typed-rpc-cli/src/update-main-schema.js' {
  declare module.exports: $Exports<'@uber/typed-rpc-cli/src/update-main-schema'>;
}
