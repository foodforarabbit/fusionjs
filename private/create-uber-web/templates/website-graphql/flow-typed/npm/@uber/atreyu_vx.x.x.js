// flow-typed signature: 780b41404a373c25f2374efbef04a942
// flow-typed version: <<STUB>>/@uber/atreyu_v^6.0.0/flow_v0.90.0

/**
 * This is an autogenerated libdef stub for:
 *
 *   '@uber/atreyu'
 *
 * Fill this stub out by replacing all the `any` types.
 *
 * Once filled out, we encourage you to share your work with the
 * community by sending a pull request to:
 * https://github.com/flowtype/flow-typed
 */

declare module '@uber/atreyu' {
  declare module.exports: any;
}

/**
 * We include stubs for each file inside this npm package in case you need to
 * require those files directly. Feel free to delete any files that aren't
 * needed.
 */
declare module '@uber/atreyu/examples/atreyu' {
  declare module.exports: any;
}

declare module '@uber/atreyu/examples/graph' {
  declare module.exports: any;
}

declare module '@uber/atreyu/examples/http-tchannel-thrift/http-tchannel-thrift' {
  declare module.exports: any;
}

declare module '@uber/atreyu/examples/http-tchannel-thrift/thrift/gen-nodejs/riders_types' {
  declare module.exports: any;
}

declare module '@uber/atreyu/examples/http-tchannel-thrift/thrift/gen-nodejs/Riders' {
  declare module.exports: any;
}

declare module '@uber/atreyu/examples/populous/populous' {
  declare module.exports: any;
}

declare module '@uber/atreyu/examples/request' {
  declare module.exports: any;
}

declare module '@uber/atreyu/lib/atreyu' {
  declare module.exports: any;
}

declare module '@uber/atreyu/lib/helpers/hyperbahn' {
  declare module.exports: any;
}

declare module '@uber/atreyu/lib/helpers/tchannel' {
  declare module.exports: any;
}

declare module '@uber/atreyu/test/atreyu.tap' {
  declare module.exports: any;
}

declare module '@uber/atreyu/test/hyperbahn.tap' {
  declare module.exports: any;
}

declare module '@uber/atreyu/test/index' {
  declare module.exports: any;
}

declare module '@uber/atreyu/test/tchannel.tap' {
  declare module.exports: any;
}

declare module '@uber/atreyu/test/thrift/gen-nodejs/helloworld_types' {
  declare module.exports: any;
}

declare module '@uber/atreyu/test/thrift/gen-nodejs/HelloWorld' {
  declare module.exports: any;
}

// Filename aliases
declare module '@uber/atreyu/examples/atreyu.js' {
  declare module.exports: $Exports<'@uber/atreyu/examples/atreyu'>;
}
declare module '@uber/atreyu/examples/graph.js' {
  declare module.exports: $Exports<'@uber/atreyu/examples/graph'>;
}
declare module '@uber/atreyu/examples/http-tchannel-thrift/http-tchannel-thrift.js' {
  declare module.exports: $Exports<'@uber/atreyu/examples/http-tchannel-thrift/http-tchannel-thrift'>;
}
declare module '@uber/atreyu/examples/http-tchannel-thrift/thrift/gen-nodejs/riders_types.js' {
  declare module.exports: $Exports<'@uber/atreyu/examples/http-tchannel-thrift/thrift/gen-nodejs/riders_types'>;
}
declare module '@uber/atreyu/examples/http-tchannel-thrift/thrift/gen-nodejs/Riders.js' {
  declare module.exports: $Exports<'@uber/atreyu/examples/http-tchannel-thrift/thrift/gen-nodejs/Riders'>;
}
declare module '@uber/atreyu/examples/populous/populous.js' {
  declare module.exports: $Exports<'@uber/atreyu/examples/populous/populous'>;
}
declare module '@uber/atreyu/examples/request.js' {
  declare module.exports: $Exports<'@uber/atreyu/examples/request'>;
}
declare module '@uber/atreyu/index' {
  declare module.exports: $Exports<'@uber/atreyu'>;
}
declare module '@uber/atreyu/index.js' {
  declare module.exports: $Exports<'@uber/atreyu'>;
}
declare module '@uber/atreyu/lib/atreyu.js' {
  declare module.exports: $Exports<'@uber/atreyu/lib/atreyu'>;
}
declare module '@uber/atreyu/lib/helpers/hyperbahn.js' {
  declare module.exports: $Exports<'@uber/atreyu/lib/helpers/hyperbahn'>;
}
declare module '@uber/atreyu/lib/helpers/tchannel.js' {
  declare module.exports: $Exports<'@uber/atreyu/lib/helpers/tchannel'>;
}
declare module '@uber/atreyu/test/atreyu.tap.js' {
  declare module.exports: $Exports<'@uber/atreyu/test/atreyu.tap'>;
}
declare module '@uber/atreyu/test/hyperbahn.tap.js' {
  declare module.exports: $Exports<'@uber/atreyu/test/hyperbahn.tap'>;
}
declare module '@uber/atreyu/test/index.js' {
  declare module.exports: $Exports<'@uber/atreyu/test/index'>;
}
declare module '@uber/atreyu/test/tchannel.tap.js' {
  declare module.exports: $Exports<'@uber/atreyu/test/tchannel.tap'>;
}
declare module '@uber/atreyu/test/thrift/gen-nodejs/helloworld_types.js' {
  declare module.exports: $Exports<'@uber/atreyu/test/thrift/gen-nodejs/helloworld_types'>;
}
declare module '@uber/atreyu/test/thrift/gen-nodejs/HelloWorld.js' {
  declare module.exports: $Exports<'@uber/atreyu/test/thrift/gen-nodejs/HelloWorld'>;
}
