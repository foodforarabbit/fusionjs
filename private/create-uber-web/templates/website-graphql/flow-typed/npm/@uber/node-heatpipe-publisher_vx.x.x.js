// flow-typed signature: c3742bb871b6f4aa4f0fc4ccc76cde7e
// flow-typed version: <<STUB>>/@uber/node-heatpipe-publisher_v^3.0.1/flow_v0.90.0

/**
 * This is an autogenerated libdef stub for:
 *
 *   '@uber/node-heatpipe-publisher'
 *
 * Fill this stub out by replacing all the `any` types.
 *
 * Once filled out, we encourage you to share your work with the
 * community by sending a pull request to:
 * https://github.com/flowtype/flow-typed
 */

declare module '@uber/node-heatpipe-publisher' {
  declare module.exports: any;
}

/**
 * We include stubs for each file inside this npm package in case you need to
 * require those files directly. Feel free to delete any files that aren't
 * needed.
 */
declare module '@uber/node-heatpipe-publisher/example' {
  declare module.exports: any;
}

declare module '@uber/node-heatpipe-publisher/lib/heatpipe_publisher' {
  declare module.exports: any;
}

declare module '@uber/node-heatpipe-publisher/test/index' {
  declare module.exports: any;
}

declare module '@uber/node-heatpipe-publisher/test/test_heatpipe_publisher' {
  declare module.exports: any;
}

// Filename aliases
declare module '@uber/node-heatpipe-publisher/example.js' {
  declare module.exports: $Exports<'@uber/node-heatpipe-publisher/example'>;
}
declare module '@uber/node-heatpipe-publisher/index' {
  declare module.exports: $Exports<'@uber/node-heatpipe-publisher'>;
}
declare module '@uber/node-heatpipe-publisher/index.js' {
  declare module.exports: $Exports<'@uber/node-heatpipe-publisher'>;
}
declare module '@uber/node-heatpipe-publisher/lib/heatpipe_publisher.js' {
  declare module.exports: $Exports<'@uber/node-heatpipe-publisher/lib/heatpipe_publisher'>;
}
declare module '@uber/node-heatpipe-publisher/test/index.js' {
  declare module.exports: $Exports<'@uber/node-heatpipe-publisher/test/index'>;
}
declare module '@uber/node-heatpipe-publisher/test/test_heatpipe_publisher.js' {
  declare module.exports: $Exports<'@uber/node-heatpipe-publisher/test/test_heatpipe_publisher'>;
}
