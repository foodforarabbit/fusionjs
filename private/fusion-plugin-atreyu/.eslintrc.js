module.exports = {
  extends: [
    require.resolve('eslint-config-uber-universal-stage-3')
  ]
};
