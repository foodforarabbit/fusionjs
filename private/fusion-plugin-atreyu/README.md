# Atreyu plugin for fusion

This is a fusion plugin for [Atreyu](https://code.uberinternal.com/diffusion/WEATREY/browse/master/#what-is-atreyu)

Atreyu is data querying/aggregation library similar to GraphQL, but designed to be used with Uber's micro-service infrastructure.

```js
// src/main.js
import App from '@fusion/react';
import Atreyu from '@uber/fusion-plugin-atreyu';
import {config} from './atreyu/config';

export default () => {
  const app = new App();
  const Atreyu = app.plugin(AtreyuPlugin, {config, M3, Logger, Tracer, Galileo})
  // atreyu client
  const atreyu = Atreyu.of();
}

// src/atreyu/config.js
export default {
  config: {
    serviceNames: ["safari"],
  },
}

#### Plugin registration

```
const Services = app.plugin(Atreyu, {config, M3, Logger, Tracer, Galileo, TChannel})
```

- `config: {serviceNames, services}` - an Atreyu config (see https://code.uberinternal.com/diffusion/WEATREY/)
- `M3` - m3 plugin
- `Logger` - logger plugin
- `Tracer` - tracer plugin
- `Galileo` - galileo plugin
- `TChannel` - tchannel plugin
