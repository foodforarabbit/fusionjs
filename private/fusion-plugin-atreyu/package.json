{
  "name": "@uber/fusion-plugin-atreyu",
  "version": "0.2.0",
  "description": "A fusion plugin for atreyu",
  "main": "./dist/node.cjs.js",
  "module": "./dist/node.es.js",
  "files": [
    "dist"
  ],
  "browser": {
    "./dist/node.cjs.js": "./dist/browser.es5.cjs.js",
    "./dist/node.es.js": "./dist/browser.es5.es.js"
  },
  "es2015": {
    "./dist/browser.es5.cjs.js": "./dist/browser.es2015.cjs.js",
    "./dist/browser.es5.es.js": "./dist/browser.es2015.es.js"
  },
  "es2017": {
    "./dist/browser.es5.cjs.js": "./dist/browser.es2017.cjs.js",
    "./dist/browser.es5.es.js": "./dist/browser.es2017.es.js",
    "./dist/browser.es2015.cjs.js": "./dist/browser.es2017.cjs.js",
    "./dist/browser.es2015.es.js": "./dist/browser.es2017.es.js"
  },
  "peerDependencies": {
    "@uber/atreyu": "^5.4.5",
    "@uber/fusion-plugin-m3": "0.3.0",
    "@uber/fusion-plugin-tracer": "0.2.0",
    "@uber/fusion-plugin-galileo": "0.2.1",
    "@uber/fusion-plugin-tchannel": "0.3.0",
    "fusion-core": "^0.3.0-5",
    "fusion-tokens": "0.0.5"
  },
  "devDependencies": {
    "@uber/fusion-plugin-m3": "0.3.0",
    "@uber/fusion-plugin-tracer": "0.2.0",
    "@uber/fusion-plugin-galileo": "0.2.0",
    "@uber/fusion-plugin-tchannel": "0.3.0",
    "babel-eslint": "8.0.2",
    "babel-plugin-istanbul": "^4.1.5",
    "create-universal-package": "^2.1.1",
    "eslint": "^4.11.0",
    "eslint-config-fusion": "^0.1.2",
    "eslint-plugin-cup": "^1.0.0",
    "eslint-plugin-flowtype": "^2.39.1",
    "eslint-plugin-prettier": "^2.3.1",
    "eslint-plugin-react": "^7.4.0",
    "fusion-core": "^0.3.0-5",
    "fusion-tokens": "0.0.5",
    "idl": "^3.1.9",
    "nyc": "^11.3.0",
    "prettier": "1.8.2",
    "tape-cup": "^4.7.1",
    "unitest": "^1.1.0"
  },
  "scripts": {
    "lint": "eslint src/",
    "clean": "rm -rf dist",
    "transpile": "npm run clean && cup build",
    "build-test": "rm -rf dist-tests && NODE_ENV=test cup build-tests",
    "just-test": "unitest --browser=dist-tests/browser.js --node=dist-tests/node.js",
    "test": "npm run lint && npm run build-test && npm run just-test",
    "cover": "npm run build-test && npm run just-cover",
    "just-cover": "nyc --reporter=cobertura npm run just-test",
    "view-cover": "npm run build-test && nyc --reporter=html npm run just-test && open coverage/index.html",
    "prepublish": "npm run transpile"
  },
  "dependencies": {
    "@uber/atreyu": "^5.4.5",
    "@uber/atreyu-test": "^1.3.0"
  },
  "engines": {
    "node": ">=8.3.0 <9.0.0",
    "npm": ">=5.0.0",
    "yarn": ">=1.0.0"
  }
}
