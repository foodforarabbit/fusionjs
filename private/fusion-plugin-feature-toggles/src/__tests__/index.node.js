// @flow
/* eslint-env node */

import {JSDOM} from 'jsdom';

import {getService, getSimulator} from 'fusion-test-utils';
import App from 'fusion-core';
import type {Context} from 'fusion-core';
import {AtreyuToken} from '@uber/fusion-plugin-atreyu';

import FeatureTogglesPlugin from '../server.js';

import {
  FeatureTogglesClientToken,
  FeatureTogglesTogglesConfigToken,
} from '../index.js';
import type {
  ToggleDetailsType,
  IFeatureTogglesClient,
  FeatureToggleConfigType,
} from '../index.js';

type AtreyuType = any;

/* Mock(s) */
const mockContextFactory: () => Context = () =>
  ({
    // necessary to handle ctx.memoized call
    memoized: new Map(),
  }: any);
const mockAtreyuFactory: () => AtreyuType = () => ({}: any);
type MockDataType = {[string]: ToggleDetailsType};
const mockClientFactory: (
  data?: MockDataType
) => Class<IFeatureTogglesClient> = data => {
  class MockClient implements IFeatureTogglesClient {
    constructor(ctx: Context): IFeatureTogglesClient {
      return this;
    }
    async load(): Promise<void> {}
    get(toggleName: string): ToggleDetailsType {
      if (!data || !data[toggleName]) return {enabled: false};
      return data[toggleName];
    }
  }
  return MockClient;
};

/* App Creator(s) */
const appCreator = (
  data?: MockDataType,
  toggleConfigs?: ?Array<FeatureToggleConfigType | string>
) => () => {
  const app = new App('el', el => el);

  app.register(FeatureTogglesPlugin);
  app.register(AtreyuToken, mockAtreyuFactory());
  app.register(FeatureTogglesClientToken, mockClientFactory(data));
  app.register(
    FeatureTogglesTogglesConfigToken,
    toggleConfigs ? toggleConfigs : ['weatherToggle']
  );

  return app;
};

test('simple service sanity check - toggle on/off with mocked dependencies', async () => {
  const mockContext = mockContextFactory();
  const mockToggles = {
    toggleOn: {
      enabled: true,
    },
    toggleOff: {
      enabled: false,
    },
  };
  const service = getService(appCreator(mockToggles), FeatureTogglesPlugin);
  const instance = service.from(mockContext);

  const onResult = await instance.get('toggleOn');
  expect(onResult).not.toBeNull();
  if (!onResult) throw new Error('onResult is null!'); // necessary to appease Flow
  expect(onResult).toHaveProperty('enabled');
  expect(onResult.enabled).toBeTruthy();

  const offResult = await instance.get('toggleOff');
  expect(offResult).not.toBeNull();
  if (!offResult) throw new Error('offResult is null!'); // necessary to appease Flow
  expect(offResult).toHaveProperty('enabled');
  expect(offResult.enabled).not.toBeTruthy();

  const missingResult = await instance.get('missingToggle');
  expect(missingResult).not.toBeNull();
  if (!missingResult) throw new Error('missingResult is null!'); // necessary to appease Flow
  expect(missingResult).toHaveProperty('enabled');
  expect(missingResult.enabled).not.toBeTruthy();
});

test('test service metadata with mocked dependencies', async () => {
  const mockContext = mockContextFactory();
  const mockToggles = {
    weatherToggle: {
      enabled: true,
      metadata: {
        isItSunnyToday: true,
        areThereWaves: false,
      },
    },
  };
  const service = getService(appCreator(mockToggles), FeatureTogglesPlugin);
  const instance = service.from(mockContext);

  const result = await instance.get('weatherToggle');
  expect(result).not.toBeNull();
  if (!result) throw new Error('Result is null!'); // necessary to appease Flow
  expect(result).toHaveProperty('metadata');
  const metadata = result.metadata;
  expect(metadata).not.toBeNull();
  if (!metadata) throw new Error('Metadata is null!'); // necessary to appease Flow
  expect(metadata).toHaveProperty('isItSunnyToday');
  expect(metadata.isItSunnyToday).toBeTruthy();
  expect(metadata).toHaveProperty('areThereWaves');
  expect(metadata.areThereWaves).not.toBeTruthy();
});

test('test ssr', async () => {
  const mockContext = mockContextFactory();
  const mockToggles = {
    weatherToggle: {
      enabled: true,
      metadata: {
        isItSunnyToday: true,
        areThereWaves: false,
      },
    },
  };
  const service = getService(appCreator(mockToggles), FeatureTogglesPlugin);
  const instance = service.from(mockContext);

  const result = await instance.get('weatherToggle');
  expect(result).not.toBeNull();
  if (!result) throw new Error('Result is null!'); // necessary to appease Flow
  expect(result).toHaveProperty('metadata');
  const metadata = result.metadata;
  expect(metadata).not.toBeNull();
  if (!metadata) throw new Error('Metadata is null!'); // necessary to appease Flow
  expect(metadata).toHaveProperty('isItSunnyToday');
  expect(metadata.isItSunnyToday).toBeTruthy();
  expect(metadata).toHaveProperty('areThereWaves');
  expect(metadata.areThereWaves).not.toBeTruthy();
});

test('test default client dependencies', async () => {
  const appFactory = () => {
    const app = new App('el', el => el);
    app.register(FeatureTogglesPlugin);
    app.register(FeatureTogglesTogglesConfigToken, []);
    return app;
  };

  // Without required (for default) AtreyuToken registration
  expect(() => getService(appFactory, FeatureTogglesPlugin)).toThrow();

  // With Atrey client
  const app = appFactory();
  app.register(AtreyuToken, mockAtreyuFactory());
  expect(() => getService(() => app, FeatureTogglesPlugin)).not.toThrow();
});

test('__FEATURE_TOGGLES__ populated', async () => {
  const mockToggles = {
    weatherToggle: {
      enabled: true,
      metadata: {
        isItSunnyToday: true,
        areThereWaves: false,
      },
    },
  };
  const app = appCreator(mockToggles)();
  const result = await getSimulator(app).render('/');

  const dom = new JSDOM(result.body);
  const elem = dom.window.document.querySelector(
    "script[id='__FEATURE_TOGGLES__'"
  );
  expect(elem).toBeTruthy();
  expect(elem).toHaveProperty('outerHTML');
  expect(elem.outerHTML).toMatchSnapshot('__FEATURE_TOGGLES__');
});

test('context is required on server', async () => {
  const service = getService(appCreator(), FeatureTogglesPlugin);
  expect(() => service.from()).toThrow();
});

test('only client-safe toggles exposed in __FEATURE_TOGGLES__ ', async () => {
  const mockToggles = {
    weatherToggle: {
      enabled: true,
      metadata: {
        isItSunnyToday: true,
        areThereWaves: false,
      },
    },
    serverOnly: {
      enabled: true,
      metadata: {
        shouldNotBeOnClient: false,
      },
    },
  };

  const app = appCreator(mockToggles, [
    {name: 'serverOnly', exposeToClient: false},
    'weatherToggle',
  ])();
  const result = await getSimulator(app).render('/');

  const dom = new JSDOM(result.body);
  const elem = dom.window.document.querySelector(
    "script[id='__FEATURE_TOGGLES__'"
  );
  expect(elem).toBeTruthy();
  expect(elem).toHaveProperty('outerHTML');
  expect(elem.outerHTML).toMatchSnapshot(
    '__FEATURE_TOGGLES__-limited-client-side'
  );
});

test('skip middleware if no toggle details provided', async () => {
  const app = appCreator({}, [])();

  // Override 'FeatureTogglesClientToken' registration to track .load call
  const loadFn = jest.fn();
  const getFn = jest.fn();
  class MockCheckLoadClient implements IFeatureTogglesClient {
    constructor(ctx: Context): IFeatureTogglesClient {
      return this;
    }
    async load(): Promise<void> {
      return loadFn();
    }
    get(toggleName: string): ToggleDetailsType {
      return getFn();
    }
  }
  app.register(FeatureTogglesClientToken, MockCheckLoadClient);

  const result = await getSimulator(app).render('/');

  // Ensure load and get were not called
  expect(loadFn).toHaveBeenCalledTimes(0);
  expect(getFn).toHaveBeenCalledTimes(0);

  const dom = new JSDOM(result.body);
  const elem = dom.window.document.querySelector(
    "script[id='__FEATURE_TOGGLES__'"
  );
  expect(elem).toBeTruthy();
  expect(elem).toHaveProperty('outerHTML');
  expect(elem.outerHTML).toMatchSnapshot(
    '__FEATURE_TOGGLES__-empty-no-details-provided'
  );
});
