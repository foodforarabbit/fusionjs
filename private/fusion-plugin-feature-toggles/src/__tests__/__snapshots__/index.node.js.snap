// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`__FEATURE_TOGGLES__ populated: __FEATURE_TOGGLES__ 1`] = `
"<script type=\\"application/json\\" id=\\"__FEATURE_TOGGLES__\\">
      {\\\\u0022data\\\\u0022:{\\\\u0022weatherToggle\\\\u0022:{\\\\u0022enabled\\\\u0022:true,\\\\u0022metadata\\\\u0022:{\\\\u0022isItSunnyToday\\\\u0022:true,\\\\u0022areThereWaves\\\\u0022:false}}}}
    </script>"
`;

exports[`only client-safe toggles exposed in __FEATURE_TOGGLES__ : __FEATURE_TOGGLES__-limited-client-side 1`] = `
"<script type=\\"application/json\\" id=\\"__FEATURE_TOGGLES__\\">
      {\\\\u0022data\\\\u0022:{\\\\u0022weatherToggle\\\\u0022:{\\\\u0022enabled\\\\u0022:true,\\\\u0022metadata\\\\u0022:{\\\\u0022isItSunnyToday\\\\u0022:true,\\\\u0022areThereWaves\\\\u0022:false}}}}
    </script>"
`;

exports[`skip middleware if no toggle details provided: __FEATURE_TOGGLES__-empty-no-details-provided 1`] = `
"<script type=\\"application/json\\" id=\\"__FEATURE_TOGGLES__\\">
      {\\\\u0022data\\\\u0022:{}}
    </script>"
`;
